src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPASITY = 10;
+    private static final int NUMBER_ONE = 1;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPASITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increaseArraySizeIfNecessary(size);
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        increaseArraySizeIfNecessary(size);
 
+        T[] secondArrayPart = getArrayCopy(array, index, array.length);
+        array[index] = value;
+
+        int startIndexForCopyArray = index + NUMBER_ONE;
+        System.arraycopy(secondArrayPart, 0, array, startIndexForCopyArray,
+                secondArrayPart.length - startIndexForCopyArray);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        size = size + list.size();
+        increaseArraySizeIfNecessary(size);
 
+        int startIndexForListElements = size - list.size();
+        for (int i = 0; i < list.size(); i++) {
+            array[startIndexForListElements + i] = list.get(i);
+        }
```

Comment:
call add in a loop
maybe multiple resizes will be needed - it will be handled in add